Composer/packages/lib/shared/__tests__/copyUtils/copyAdaptiveAction.test.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { copyAdaptiveAction } from '../../src/copyUtils';
+import { CopyConstructorMap } from '../../src/copyUtils/copyAdaptiveAction';
+import { externalApiStub as externalApi } from '../jestMocks/externalApiStub';
+import { SDKTypes } from '../../src';
+
+describe('copyAdaptiveAction', () => {
+  it('should return {} when input is invalid', async () => {
+    expect(await copyAdaptiveAction('hello', externalApi)).toEqual('hello');
+
+    expect(await copyAdaptiveAction(null as any, externalApi)).toEqual({});
+    expect(await copyAdaptiveAction({} as any, externalApi)).toEqual({});
+    expect(await copyAdaptiveAction({ name: 'hi' } as any, externalApi)).toEqual({});
+  });
+
+  describe('should call certain handler in CopyConstructorMap', () => {
```

Comment:
From a test organization point of view, this is not very descriptive.

here's my suggestion for this file:

```
describe('copyAdaptiveAction')
  it('should return {} when input is invalid')
 
  for type in registeredUniqueTypes
    it(`should invoke correct handler for ${type}')

  it('should invoke default handler for other types')
```

Reply:
great! I will accept this one